### PR TITLE
Report data model (aka javascript) compilation errors before the FSM is executed

### PR DESCRIPTION
**Pull request by mattoshry**
This behavior is only enabled when the reportAllErrors option is explicitly set to true in the xxxToModel APIs.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 language: node_js
 node_js:
+  - stable
   - 0.12
   - 0.11
   - 0.10
-  - 0.8
```
Comment (jbeard4) [inline .travis.yml]: 👍

File: README.md [hunk 2]
```diff
@@ -76,10 +76,10 @@ Note that this will only work if the SCXML `<script>` tags contain local filesys
...
 ## Instantiation
 
-### scxml.urlToModel(url,function(err, model){})
-### scxml.pathToModel(path,function(err, model){})
-### scxml.documentStringToModel(scxmlDocString,function(err, model){})
-### scxml.documentToModel(scxmlDocument,function(err, model){})
+### scxml.urlToModel(url,function(err, model){} [, context])
+### scxml.pathToModel(path,function(err, model){} [, context])
+### scxml.documentStringToModel(scxmlDocString,function(err, model){} [, context])
+### scxml.documentToModel(scxmlDocument,function(err, model){} [, context])
```
Comment (jbeard4) [inline README.md]: 👍

File: bin/repl.js [hunk 3]
```diff
@@ -31,15 +46,17 @@ scxml.pathToModel(pathToScxml,function(err,model){
...
     console.log(interpreter.getConfiguration());
 
-    var parseRE = /\((.*)\n\)/;
-
     function processEvent(cmd,dontKnow,alsoDontKnow,callback){
-        var e = cmd.match(parseRE)[1];
-        interpreter.gen({name : e});
+        cmd = cmd.trim();
+        interpreter.gen({name : cmd});
         var conf = interpreter.getConfiguration();
```
Comment (jbeard4) [inline bin/repl.js]: Can you briefly describe the motivation for this?

Comment (mattoshry) [inline bin/repl.js]: It seemed cumbersome to require each fsm input to be surrounded by parens. If it's important I can restore it and add an explanatory comment that you provide.

**Comment by jbeard4** [comment]
This looks good. Could you please submit a platform-test that exercises context.reportAllErrors? 

Right now, there are two forms of tests in SCION:
  * tests that check compliance with the SCXML specification. These are located in test/scxml-test-framework
  * tests that check features specific to the SCION API. These are located in test/platform-tests

You can use https://github.com/jbeard4/SCION/tree/cleanup-platform-tests/test/platform-tests/node/require as an example of how to write a test.

Please put the new test in:
test/platform-tests/node/catch-script-syntax-errors-on-compile/

This test can have any format, but please ensure that the directory contains a nodeunit test script called runner.js

I have added preliminary automation support for travis. You can pull from my branch to get these changes: https://github.com/jbeard4/SCION/tree/cleanup-platform-tests

If you have any questions, please don't hesitate to ask. Thank.

**Comment by mattoshry** [comment]
Added installation of nodeunit to run-tests-node.sh to fix the build 430 failure. build 431 passes on 3/4 platforms. It fails on node@0.11 due scxml-test-framework/test/parallel+interrupt/test24.scxml. Please take a look.

**Comment by jbeard4** [comment]
Will review this evening. Thanks.

**Comment by mattoshry** [comment]
I kicked the build, and this time node@0.11 passed. The other versions passed as well.

**Pull request merged**